### PR TITLE
Split the devbox image from /Dockerfile into /docker/dev/Dockerfile

### LINKED ISSUE #2196: docker build command often rebuild the image
Currently, we have only one Dockerfile, which defines the development environment and the released image.

This approach has a problem -- while I am developing in the local Git repo, I often change some files.  When I run `docker build`, the changed file would trigger the rebuilding process, which takes a long time.

Also, the generated Docker image is huge.

It looks like a good idea to separate the development environment image and the released image.

So, `docker build -t sqlflow:dev .` should not depend on the SQLFlow source code.  And `docker run --rm $PWD:/work sqlflow:dev` should generate a directory of built binaries and a Dockerfile, like

```
build/
 |- *.jar - Java parsers
 |- sqlflowserver - Go binary
 |- sqlflow - Go binary
 |- step - Go binary
 |- python/sqlflow - Python code
 |- notebook - Jupyter notebooks converted from Markdown files
```

Then, by running `cd build && docker build -t sqlflow .`, we get the released image.

### PR DESCRIPTION
**Pull request by wangkuiyi**
Fix https://github.com/sql-machine-learning/sqlflow/issues/2196

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -31,93 +10,31 @@ COPY doc/datasets/popularize_churn.sql \
...
+ENV PYTHONPATH=/usr/local/sqlflow/python:$PYTHONPATH
 
-# The SQLFlow magic command for Jupyter.
-ENV IPYTHON_STARTUP /root/.ipython/profile_default/startup/
-COPY scripts/docker/install-jupyter.bash /
-COPY scripts/docker/js /js
-RUN /install-jupyter.bash
+# Install Couler, Fluid, and model zoo.
+COPY build/*.whl /usr/local/sqlflow/python/
+RUN pip install /usr/local/sqlflow/python/*.whl
```
Comment (typhoonzero) [inline Dockerfile:20]: `RUN pip install /usr/local/sqlflow/python/*.whl && rm -f /usr/local/sqlflow/python/*.whl` to make the image smaller

File: scripts/travis/build.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
Comment (typhoonzero) [inline scripts/travis/build.sh:1]: build.sh -> build_images.sh?

File: scripts/travis/push.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
Comment (typhoonzero) [inline scripts/travis/push.sh:1]: push.sh -> push_images.sh?

File: docker/dev/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+# The SQLFlow magic command for Jupyter.
+ENV IPYTHON_STARTUP /root/.ipython/profile_default/startup/
+COPY install-jupyter.bash /
+COPY js /js
+RUN /install-jupyter.bash
+
+# Set build.sh as the entrypoint and assume that SQLFlow source tree
+# is at /work.
+COPY build.sh /
+CMD ["/build.sh", "/work"]
```
Comment (typhoonzero) [inline docker/dev/Dockerfile:72]: The dev box is just some environment that needed by SQLFlow building and developing, so maybe remove here, and do `docker run --rm -v /path/to/sqlflow:/go/src/sqlflow.org/sqlflow sqlflow/sqlflow:dev /go/src/sqlflow.org/sqlflow/scripts/build.sh` ?

**Review by Yancey0623 — APPROVED** [review]
LGTM

**Comment by wangkuiyi** [comment]
According to the records on DockerHub.com

Screen Shot 2020-05-10 at 7 22 17 PM

before this change, we have the all-in-one Docker image `sqlflow/sqlflow:latest` of size 2.5GB; after this change, we have two images, `sqlflow/sqlflow:dev` takes 1.89 GB, and the new all-in-one image `sqlflow/sqlflow:ci` is 2.11 GB and is smaller than the old one.

**Pull request merged**